libsel4/tools/generator.py

```diff
@@ -0,0 +1,612 @@
+#
+# Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+
+from functools import reduce
+import itertools
+import operator
+import xml.dom.minidom
+
+
+# Maximum number of words that will be in a message.
+MAX_MESSAGE_LENGTH = 64
+
+# Number of bits in a standard word
+WORD_SIZE_BITS_ARCH = {
```

Comment:
Looking at this I wonder if it might scale better if there is an "arch object" that can tell the bit size, the number of message registers and maybe more.

Reply:
Yes, right now the information about architecture is handled differently, there's `Architecture` class for that.